client/client.go

```diff
@@ -115,13 +182,30 @@ func (client *Client) writeLoop() {
 
 		binary.BigEndian.PutUint32(ibuf, length)
 
-		client.conn.Write(ibuf)
+		if _, err := client.getConn().Write(ibuf); err != nil {
+			if err = client.reconnect(err); err != nil {
+				break
+			}
+		}
 
-		client.conn.Write(req.data[0])
+		if _, err := client.getConn().Write(req.data[0]); err != nil {
```

Comment:
Can you factor this out into a function? It's repeated many times. Something like
```
(client * Client) WriteBytes(bytes []byte) err{
		if _, err := client.getConn().Write(bytes); err != nil {
			if err = client.reconnect(err); err != nil {
                                   return err
			}
		}
}

Reply:
refactored to use a one-liner